consensus/hotstuff/signer.go

```diff
@@ -2,23 +2,13 @@ package hotstuff
 
 import (
 	"github.com/onflow/flow-go/consensus/hotstuff/model"
-	"github.com/onflow/flow-go/model/flow"
 )
 
-// SignerVerifier can sign and verify HotStuff entities.
-type SignerVerifier interface {
-	Signer
-	Verifier
-}
-
-// Signer is responsible for creating votes, proposals and QC's for a given block.
+// Signer is responsible for creating votes, proposals for a given block.
 type Signer interface {
```

Comment:
- I'm a bit confused between this `Signer` and `Voter` under consensus/hotstuff/voter.go. Both seem to be producing votes..
- There in another `Signer` under module/signer.go that we can delete too.

Reply:
`Voter` uses `Signer` to sign vote, and `Voter` also has voting rule in it. 

>There in another Signer under module/signer.go that we can delete too.

Will be done in another clean up PR

Reply:
Thanks Leo!
This is low priority: 
I see the difference, the `Signer` still sees some voting context (block, checks for proposal identity..) so it does a bit more than just signing. The main signing function is `genSigData` [here](https://github.com/onflow/flow-go/blob/leo/replace-signer/consensus/hotstuff/verification/combined_signer_v2.go#L98). Btw, there are 2 structures implementing `Signer`: `CombinedSigner` and `StakingSigner`. Both have exactly the same code, but are only different in `genSigData`.

I think a cleaner design would be :
```
type Signer {
    GenSigData
 }
```
- `Signer` is implemented differently by `CombinedSigner` and `StakingSigner`.
- `Voter` needs an instance of `Signer` to blindly sign for votes. All voting logic is inside `Voter`.
- `BlockProducer` needs an instance of `Signer` to blindly sign for proposals. All proposal logic is inside `BlockProducer`.
-  There is no intermediate interface that has methods `CreateProposal` and `CreateVote`.

Reply:
This is a really good point. I like your idea. I have [issue it up](https://github.com/dapperlabs/flow-go/issues/6055), as the changes for this refactor might not be big.